velox/common/base/AbstractBloomFilter.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+class AbstractBloomFilter {
+ public:
+  virtual bool mightContain(int32_t value) const = 0;
```

Comment:
Let's remove this method and just use `int64_t` version.  `BloomFilter::hash(int32_t)` and `BloomFilter::hash(int64_t)` should return the same value, otherwise it's dangerous.  You probably want to delete `parquet::BloomFilter::hash(int32_t)` as well.

Reply:
The computed hash value for `int32` and `int64` is different. I'm testing with the parquet file created using Spark. If we use `int64` version, then the bloom filter check fails.

Reply:
> The computed hash value for int32 and int64 is different

Then relying on overloaded methods is very dangerous.  Let's use different names for the methods:
- `mightContainInt64`, `mightContainInt32`
- `hashInt64`, `hashInt32`, `hashFloat`, `hashDouble`